tool/src/c2/ty.rs

```diff
@@ -6,6 +6,10 @@ use std::fmt::Write;
 
 impl<'tcx> super::CContext<'tcx> {
     pub fn gen_ty(&self, id: TypeId, ty: TypeDef<'tcx>) {
+        if ty.attrs().disable {
```

Comment:
thought: this happens quite late and every backend will have to make sure it checks for disabled items in several places. Is there an earlier stage where they could be removed?

Reply:
This was a deliberate choice that I discussed with @sffc. I'd like backends to handle backend attributes, rather than HIR lowering. `disable`, specifically, _could_ be handled by lowering, I felt it would be more consistent to handle them all in the backends, even if it's more work for this one specifically.

(Also the error handling gets more consistent.